packages/react/src/__tests__/ReactProfilerDevToolsIntegration-test.internal.js

```diff
@@ -0,0 +1,112 @@
+      onCommitFiberRoot: jest.fn((rendererId, root) => {}),
+      onCommitFiberUnmount: () => {},
+      supportsFiber: true,
+    };
+
+    jest.resetModules();
+
+    ReactFeatureFlags = require('shared/ReactFeatureFlags');
+    ReactFeatureFlags.enableProfilerTimer = true;
+    ReactFeatureFlags.supportDevToolsIfPresent = true;
+    React = require('react');
+    ReactTestRenderer = require('react-test-renderer');
+
+    mockNowForTests();
+
+    AdvanceTime = class extends React.Component {
+      static defaultProps = {
+        byAmount: 10,
+        shouldComponentUpdate: true,
+      };
+      shouldComponentUpdate(nextProps) {
+        return nextProps.shouldComponentUpdate;
+      }
+      render() {
+        // Simulate time passing when this component is rendered
+        advanceTimeBy(this.props.byAmount);
+        return this.props.children || null;
+      }
+    };
+  });
+
+  it('should auto-Profile all fibers if the DevTools hook is detected', () => {
+    const App = ({multiplier}) => {
+      advanceTimeBy(2);
+      return (
+        <React.unstable_Profiler id="Profiler" onRender={onRender}>
+          <AdvanceTime byAmount={3 * multiplier} shouldComponentUpdate={true} />
+          <AdvanceTime
+            byAmount={7 * multiplier}
+            shouldComponentUpdate={false}
+          />
+        </React.unstable_Profiler>
+      );
+    };
+
+    const onRender = jest.fn(() => {});
+    const rendered = ReactTestRenderer.create(<App multiplier={1} />);
+
+    expect(hook.onCommitFiberRoot).toHaveBeenCalledTimes(1);
+
+    // Measure observable timing using the Profiler component.
+    // The time spent in App (above the Profiler) won't be included in the durations,
+    // But needs to be acocunted for in the offset times.
+    expect(onRender).toHaveBeenCalledTimes(1);
+    expect(onRender).toHaveBeenCalledWith('Profiler', 'mount', 10, 10, 2, 12);
+    onRender.mockClear();
+
+    // Measure unobservable timing required by the DevTools profiler.
+    // At this point, the base time should include both:
+    // The time 2ms in the App component itself, and
+    // The 10ms spend in the Profiler sub-tree beneath.
+    expect(rendered.root.findByType(App)._currentFiber().treeBaseTime).toBe(12);
+
+    rendered.update(<App multiplier={2} />);
+
+    // Measure observable timing using the Profiler component.
+    // The time spent in App (above the Profiler) won't be included in the durations,
+    // But needs to be acocunted for in the offset times.
+    expect(onRender).toHaveBeenCalledTimes(1);
+    expect(onRender).toHaveBeenCalledWith('Profiler', 'update', 6, 13, 14, 20);
+
+    // Measure unobservable timing required by the DevTools profiler.
+    // At this point, the base time should include both:
+    // The initial 9ms for the components that do not re-render, and
+    // The updated 6ms for the component that does.
+    expect(rendered.root.findByType(App)._currentFiber().treeBaseTime).toBe(15);
```

Comment:
Can you explain more about why this is completely unobservable? Is it because it's only used for the UI, and not reported by the `Profiler` component? *Should* it be reported by `Profiler` component somehow?

Reply:
Maybe "unobservable" isn't the right word to use, but this time value won't be tracked at all unless the DevTools are detected– because `App` isn't inside of a `Profiler` root (which is intentional or this test).

Reply:
My main concern is with using `_currentFiber()` here which isn't a public API. Is there any way we could write these tests without reaching into the fiber data structure? If not (and if that's how DevTools works anyways) then it's okay.

Reply:
I don't think so. Also, this is kind of doing [what Sebastian suggested](https://github.com/facebook/react/pull/13058#discussion_r195890589) (assuming I understood his comment correctly).

Reply:
DevTools wouldn't be using `_currentFiber()` but yeah, this seems like the easiest way without bringing the backend itself into this repo. I assume you're reading `treeBaseTime` from the Fiber in DevTools—can't check because I'm not sure where the code is.